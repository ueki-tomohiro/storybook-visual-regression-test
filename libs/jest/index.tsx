import { composeStories, StoryFile } from "@storybook/testing-react";
import { render } from "@testing-library/react";
import React from "react";

export const renderFromStories = (stories: StoryFile) => {
  const composed = composeStories(stories);
  const testCases = Object.values(composed).map((Story: any) => [Story.storyName, Story, Story.args]);
  test.each(testCases)("renders %s", (_storyName, Component: any, args) => {
    if (_storyName.match(/Fake/)) return;

    const tree = render(<Component {...(args || {})} />);
    expect(tree.baseElement).toMatchSnapshot();
  });
};
