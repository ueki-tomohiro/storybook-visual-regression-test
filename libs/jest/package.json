{
  "name": "@demo-libs/jest",
  "version": "0.0.1",
  "private": true,
  "description": "Demo Jest",
  "license": "MIT",
  "devDependencies": {
    "@babel/core": "^7.21.4",
    "@babel/preset-env": "^7.21.4",
    "@babel/preset-typescript": "^7.21.4",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.5.1",
    "@types/jest": "^29.5.1",
    "babel-jest": "^29.5.0",
    "babel-plugin-transform-import-meta": "^2.2.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.5.0",
    "jest-environment-jsdom": "^29.5.0"
  },
  "engines": {
    "node": "^20",
    "yarn": ">=1"
  },
  "volta": {
    "node": "20.0.0",
    "yarn": "1.22.19"
  }
}
