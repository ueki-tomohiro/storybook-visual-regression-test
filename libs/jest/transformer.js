const babelJest = require("babel-jest");
const babelOptions = {
  presets: [
    [
      "@babel/env",
      {
        targets: {
          node: "current",
        },
        exclude: ["transform-regenerator"],
      },
    ],
    "@babel/preset-typescript",
    "@babel/react",
  ],
  plugins: ["babel-plugin-transform-import-meta"],
  babelrc: false,
};
module.exports = babelJest.createTransformer(babelOptions);
