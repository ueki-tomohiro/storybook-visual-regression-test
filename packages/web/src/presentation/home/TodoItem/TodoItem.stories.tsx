import { getGetTodoMock } from "@demo/api/lib/demo/todo/todo.msw";

import { TodoItem, TodoItemProps } from "./TodoItem";

const defaultArgs: TodoItemProps = {
  todo: { todo_id: 79208, description: "Bedfordshire", end_date: "2022-07-19T12:00:21+09:00", completed: false },
};

export default {
  title: "views/TodoItem",
  component: TodoItem,
};

export const Default = { args: defaultArgs };

export const Fake = {
  args: {
    todo: getGetTodoMock(),
  },
};
