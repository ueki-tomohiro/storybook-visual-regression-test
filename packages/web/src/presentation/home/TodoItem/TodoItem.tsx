import { Todo } from "@demo/api/lib/demo/model";
import dayjs from "dayjs";
import Link, { LinkProps } from "next/link";
import React from "react";

export type TodoItemProps = Omit<LinkProps, "href"> & {
  todo: Todo;
};

export const TodoItem: React.FC<TodoItemProps> = ({ todo, ...props }) => {
  return (
    <Link {...props} href={`/todo/${todo.todo_id}`} className="w-full">
      <div className="my-2 flex items-center border-2 p-2">
        <div>
          {todo.completed ? (
            <span className="material-symbols-outlined">done</span>
          ) : (
            <span className="material-symbols-outlined">label_important</span>
          )}
        </div>
        <div className="mx-2 text-xl">{todo.description}</div>
        <div className="text-tertiary text-xs">{dayjs(todo.end_date).format("YYYY-MM-DD HH:mm:ss")}</div>
      </div>
    </Link>
  );
};
