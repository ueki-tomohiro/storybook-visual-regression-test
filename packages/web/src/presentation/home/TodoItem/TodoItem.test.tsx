import { renderFromStories } from "@demo-libs/jest";

import * as stories from "./TodoItem.stories";

describe(stories.default.title, () => {
  renderFromStories(stories);
});
