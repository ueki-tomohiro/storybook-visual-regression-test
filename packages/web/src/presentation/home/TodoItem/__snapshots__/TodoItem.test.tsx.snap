// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/TodoItem renders Default 1`] = `
<body>
  <div>
    <a
      class="w-full"
      href="/todo/79208"
    >
      <div
        class="my-2 flex items-center border-2 p-2"
      >
        <div>
          <span
            class="material-symbols-outlined"
          >
            label_important
          </span>
        </div>
        <div
          class="mx-2 text-xl"
        >
          Bedfordshire
        </div>
        <div
          class="text-xs text-tertiary"
        >
          2022-07-19 12:00:21
        </div>
      </div>
    </a>
  </div>
</body>
`;
